src/test/java/uk/gov/ida/stub/idp/domain/EidasAuthnRequestTest.java

```diff
@@ -62,21 +57,32 @@ public void shouldConvertAuthnRequestToEidasAuthnRequest() {
         assertThat(actualEidasAuthnRequest.getDestination()).isEqualTo("Destination");
         assertThat(actualEidasAuthnRequest.getRequestedLoa()).isEqualTo("http://eidas.europa.eu/LoA/substantial");
 
-        XMLObject xmlObject = actualEidasAuthnRequest.getAttributes().get(0).getAttributeValues().get(0);
-        assertThat(xmlObject.getClass()).isEqualTo(CurrentFamilyNameImpl.class);
-        assertThat(((CurrentFamilyNameImpl) xmlObject).getFamilyName()).isEqualTo(currentFamilyName.getFamilyName());
+        assertThat(actualEidasAuthnRequest.getAttributes().size()).isEqualTo(1);
+        uk.gov.ida.stub.idp.domain.RequestedAttribute requestedAttribute = actualEidasAuthnRequest.getAttributes().get(0);
+        assertThat(requestedAttribute.getName()).isEqualTo(IdaConstants.Eidas_Attributes.FamilyName.NAME);
+        assertThat(requestedAttribute.isRequired()).isEqualTo(true);
     }
 
-    private Extensions anExtensionWithCurrentFamilyName(CurrentFamilyName currentFamilyName) {
-        RequestedAttribute requestedAttribute = new RequestedAttributeBuilder().buildObject();
-        requestedAttribute.getAttributeValues().add(currentFamilyName);
+    // this code is copied from EidasAuthnRequestBuilder
+    private Extensions createEidasExtensions() {
+        SPType spType = (SPType) XMLObjectSupport.buildXMLObject(SPType.DEFAULT_ELEMENT_NAME);
+        spType.setValue("public");
 
-        RequestedAttributesImpl requestedAttributes = (RequestedAttributesImpl) new RequestedAttributesBuilder().buildObject();
-        requestedAttributes.setRequestedAttributes(requestedAttribute);
+        RequestedAttributesImpl requestedAttributes = (RequestedAttributesImpl) XMLObjectSupport.buildXMLObject(RequestedAttributes.DEFAULT_ELEMENT_NAME);
+        requestedAttributes.setRequestedAttributes(createRequestedAttribute(IdaConstants.Eidas_Attributes.FamilyName.NAME));
 
-        Extensions extensions = new ExtensionsBuilder().buildObject();
+        Extensions extensions = (Extensions) XMLObjectSupport.buildXMLObject(Extensions.DEFAULT_ELEMENT_NAME);
```

Comment:
What's the reason for using `XMLObjectSupport` here instead of the builder? Feels much worse having to do a cast.

Reply:
Are the builders broken?